lib/main.coffee

```diff
@@ -1,114 +1,90 @@
-                      {scopeName} = grammar
-                      atom.grammars.removeGrammarForScopeName scopeName
-                  .then ->
-                    # Remove loaded package (Hack force reload)
-                    delete atom.packages.loadedPackages[grammarsPackageName]
-
-                    # Load package
-                    # (use `loadGrammarsSync` instead of `loadGrammars` because `loadGrammars` doesn't work properly)
-                    atom.packages.loadPackage(grammarsPackageName).loadGrammarsSync()
-                  .then =>
-                    # Reload grammars for each editor
-                    atom.workspace.getTextEditors().forEach (editor) =>
-                      if editor.getGrammar().packageName is grammarsPackageName
-                        if @debug then console.log editor.getTitle()
-                        atomVersion = parseFloat(atom.getVersion())
-                        if atomVersion < 1.11
-                          editor.reloadGrammar()
-                        else
-                          # Workaround because:
-                          # - `reloadGrammar` is buggy before 1.11 (https://github.com/atom/atom/issues/13022)
-                          # - `maintainGrammar` change this behavior and don't reload existing grammar.
-                          #   - https://github.com/atom/atom/pull/12125
-                          #   - https://github.com/atom/atom/blob/c844d0f099e6ed95c52f0b94e1f141759926aeb8/src/text-editor-registry.js#L201
-                          grammarOverride = atom.textEditors.editorGrammarOverrides[editor.id]
-                          atom.textEditors.setGrammarOverride editor, 'text.plain'
-                          if grammarOverride?
-                            atom.textEditors.setGrammarOverride editor, grammarOverride
-                          else
-                            atom.textEditors.clearGrammarOverride editor
-
-                    Promise.resolve 'success'
-              else
-                Promise.resolve projectPackage.name + ' is not the right package.'
-        .catch (error) =>
-          if @debug then console.error error
-          Promise.resolve packageJsonPath + " doesn't exists."
-
-    Promise.all(promises)
-      .then (msg) =>
-        console.log 'Grammars reloaded.'
-        if @debug then console.log msg
-      .catch (error) ->
-        console.log 'Grammars failed to reload.'
-        console.error error
-
-    deactivate: ->
-      @subscriptions?.dispose()
-      @liveReloadSubscriptions?.dispose()
+    return if atom.inSpecMode() or not atom.inDevMode()
+    @configSub = atom.config.observe 'grammar-live-reload.enabled', (enabled) =>
+      return @editorSub?.dispose() unless enabled
+      reload = @reload.bind this
+      @editorSub = atom.workspace.observeTextEditors (editor) =>
+        filePath = editor.getPath()
+        if grammarRE.test(filePath) and not @watching[filePath]
```

Comment:
I think the black list must be apply here.

Reply:
Oh that's a good point!